lib/segment/encoder.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Segment.Encoder do
+  @moduledoc """
+  Responsible to transforming structs into JSON strings.
+  """
+
+  @doc """
+  Encodes a given struct into a JSON string.
+
+  ## Options
+
+      * `drop_nil_fields`: If set to `true` all the struct `nil` fileds will be
+      filtered out from the JSON string. Defaults to `false`.
+
+  ## Examples
+
+      iex> library = %Segment.Analytics.Context.Library{
+      ...>   name: "foo",
+      ...>   version: "1.0.0"
+      ...> }
+      ...> #{inspect(__MODULE__)}.encode!(library, [])
+      ~s({"version":"1.0.0","transport":null,"name":"foo"})
+
+      iex> library = %Segment.Analytics.Context.Library{
+      ...>   name: "foo",
+      ...>   version: "1.0.0"
+      ...> }
+      ...> #{inspect(__MODULE__)}.encode!(library, drop_nil_fields: true)
+      ~s({"version":"1.0.0","name":"foo"})
+
+  """
+  @spec encode!(struct(), list()) :: String.t()
+  def encode!(struct, options) do
+    struct
+    |> maybe_drop_nil_fields(options)
+    |> Poison.encode!()
+  end
+
+  defp maybe_drop_nil_fields(struct, options) do
+    if Keyword.get(options, :drop_nil_fields) == true do
```

Comment:
Is it necessary to compare with `true`? I think `if Keyword.get(options, :drop_nil_fields) do` would have the same result.

Reply:
No necessary but if we don't compare it a `Analytics.call(something, drop_nil_fields: "Please don't")` will also enable the feature 😄